### PR TITLE
#309 View channels with the new collections

### PR DESCRIPTION
**Pull request by RkAiman**

### CODE REVIEW

File: twake/frontend/src/app/components/Leftbar/Channel/Channel.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import Emojione from 'components/Emojione/Emojione';
+import Icon from 'components/Icon/Icon.js';
+// import User from 'components/User/User.js';
+// import Draggable from 'components/Draggable/Draggable.js';
+// import Tooltip from 'components/Tooltip/Tooltip.js';
+// import Languages from 'services/languages/languages.js';
+
+type Props = {
+  name: string;
+  icon: string; // string || url || react.component
```
Comment (chamerling) [inline twake/frontend/src/app/components/Leftbar/Channel/Channel.tsx:15]: We can define an Icon type

File: twake/frontend/src/app/components/Leftbar/Channel/Channel.tsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  selected: boolean;
+  muted: boolean;
+  favorite: boolean;
+  unreadMessages: boolean;
+  visibility: string; //"private" | "public" | "direct"
```
Comment (chamerling) [inline twake/frontend/src/app/components/Leftbar/Channel/Channel.tsx:20]: We can also define a visibility Type

File: twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelsBar.js [hunk 3]
```diff
@@ -11,86 +11,92 @@ import GroupSwitch from 'components/Leftbar/GroupSwitch/GroupSwitch.js';
...
-    Workspaces.addListener(this);
-    Groups.addListener(this);
-    Collections.get('groups').addListener(this);
-    Collections.get('workspaces').addListener(this);
-    WorkspaceUserRights.addListener(this);
-  }
-  componentDidMount() {
-    this.componentDidUpdate();
-  }
+  // constructor() {
```
Comment (chamerling) [inline twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelsBar.js:26]: Even in draft, you can remove all these comments

File: twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelsUser/ChannelsUser.tsx [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+import RouterServices from 'services/RouterServices';
+import OldCollections from 'services/Depreciated/Collections/Collections';
+import { Collection } from 'services/CollectionsReact/Collections';
+import { ChannelResource } from 'app/models/Channel';
+
+export function ChannelsUser() {
+  const params: any = useParams();
+  const workspaceId: string = RouterServices.translateToUUID(params.workspaceId);
+  const companyId: string = OldCollections.get('workspaces').find(workspaceId)?.group?.id;
+  const url: string = `/companies/${companyId}/workspaces/direct/channels/`;
```
Comment (chamerling) [inline twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelsUser/ChannelsUser.tsx:31]: Types are infered, so `:string` are useless

**Comment by RomaricMourgues** [comment]
I merge this because I need it but please @RkAiman do the proposed changes

**Pull request merged**